client/me/purchases/track-purchase-page-view/index.js

```diff
@@ -0,0 +1,63 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal Dependencies
+ */
+import { getByPurchaseId } from 'state/purchases/selectors';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+export class TrackPurchasePageView extends Component {
+	static propTypes = {
+		purchaseId: PropTypes.number.isRequired,
+		productSlug: PropTypes.string,
+		recordTracksEvent: PropTypes.func.isRequired,
+		eventName: PropTypes.string.isRequired,
+	};
+
+	hasTracked = false;
+
+	trackOnce() {
+		if ( ! this.hasTracked && ( this.props.productSlug && this.props.eventName ) ) {
+			this.hasTracked = true;
+			this.props.recordTracksEvent( this.props.eventName, {
+				product_slug: this.props.productSlug,
+			} );
+		}
+	}
+
+	componentDidMount() {
+		this.trackOnce();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if (
+			this.props.eventName !== prevProps.eventName ||
+			this.props.productSlug !== prevProps.productSlug
+		) {
+			this.hasTracked = false;
+		}
+
+		this.trackOnce();
```

Comment:
I think this call could actually go in the block above, right after `this.hasTracked = false`, because the only reason that the event would not yet have fired was if one of the props had not been supplied.

It follows that the boolean is redundant, because the only time we set it to false is right before we are going to fire the tracks event.

Reply:
Good spot!